src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const sortedItemsGroup: GroupsMap<T> = {};
+
+  items.reduce((acc: GroupsMap<T>, el: T) => {
```

Comment:
Are you sure you need a `reduce` here?
You have created an `sortedItemsGroup` object where you will write information and it is enough for you to walk through the array using `forEach`.